examples/systemcalls/systemcalls.c

```diff
@@ -80,20 +226,102 @@ bool do_exec_redirect(const char *outputfile, int count, ...)
         command[i] = va_arg(args, char *);
     }
     command[count] = NULL;
-    // this line is to avoid a compile warning before your implementation is complete
-    // and may be removed
-    command[count] = command[count];
 
 
-/*
- * TODO
- *   Call execv, but first using https://stackoverflow.com/a/13784315/1446624 as a refernce,
- *   redirect standard out to a file specified by outputfile.
- *   The rest of the behaviour is same as do_exec()
- *
-*/
+    /*The path needs to be absolute so must start with root*/
+    if (command[0] == NULL || command[0][0] != '/' || outputfile == NULL)
```

Comment:
I don't think it would necessarily need to start with `/`, for instance if you were executing a file in the current working directory.